subsys/bluetooth/scan.c

```diff
@@ -283,6 +306,71 @@ static void notify_connecting_error(struct bt_scan_device_info *device_info)
 	}
 }
 
+static void attempts_filter_force_add(struct conn_attempts_filter *filter,
+				      const bt_addr_le_t *addr)
+{
+	LOG_DBG("Force adding new device filter");
+
+	/* Overwrite the oldest device */
+	memset(filter->oldest_device, 0, sizeof(*filter->oldest_device));
+	memcpy(&filter->oldest_device->addr, addr, sizeof(*addr));
```

Comment:
Maybe we could use `bt_addr_copy` and set the counter using explicit assignment here?

Reply:
I forgot about this function. Thanks